### PR TITLE
Catch the permission exception on the audio component

### PR DESCRIPTION
**Pull request by Ian-GL**
# Description

Currently the Audio component fails silently if the microphone permissions were not granted and the UI seems as if it was recording. This fix catches the permission exception and raises an alert that tells the user the permissions are needed. It also prevents the UI from showing the stop button as if it had started recording.

Screen Shot 2022-09-24 at 9 51 50

Closes: #2325 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @Ian-GL for the fix! This looks pretty good, but I wonder if it would be possible to use Gradio's Error instead of alert() for a better UI? cc @aliabid94

**Comment by Ian-GL** [comment]
> Thanks @Ian-GL for the fix! This looks pretty good, but I wonder if it would be possible to use Gradio's Error instead of alert() for a better UI? cc @aliabid94

Hey @abidlabs , I wasn't aware of the error. Yes, if you have an example of another component that uses it I could use that as a guide to add it here.

File: ui/packages/audio/src/Audio.svelte [hunk 1]
```diff
@@ -57,7 +57,26 @@
...
+			.catch((err) => {
+				if (err instanceof DOMException && err.name == "NotAllowedError") {
+					alert(
+						"Please allow the use of microphone if you wish to record audio."
+					);
+					return null;
+				} else {
+					throw err;
+				}
+			});
```
Comment (pngwn) [inline ui/packages/audio/src/Audio.svelte:74]: We shouldn't be mixing await and `then` chaining. Lets wrap this in a `try` `catch` block instead.

Additionally should think of something less intrusive than an `alert` which blocks all interaction with the page? cc @gary149 @abidlabs

**Review by pngwn — CHANGES_REQUESTED** [review]

**Comment by aliabid94** [comment]
I'll make a PR against this fork to launch a gradio.Error from within a component - we don't have any examples of doing this before so I'll take care of that.

**Comment by aliabid94** [comment]
Just made the changes to use gradio error messaging, @pngwn please rereview

**Comment by pngwn** [comment]
Pushed a change to convert the chained promises to `try` `catch` style, so we aren't mixing await + promise chaining, and fixed and issue with the type.

Looks good, thanks @Ian-GL and @aliabid94!

**Review by pngwn — APPROVED** [review]

**Comment by pngwn** [comment]
Will merge when CI passes with everything merged in.

**Comment by Ian-GL** [comment]
Thanks for taking care of the changes, @pngwn and @aliabid94 !

**Pull request merged**